contract-examples/test/utils.ts

```diff
@@ -0,0 +1,119 @@
+import { ethers } from "hardhat"
+const assert = require("assert")
+
+/*
+ *
+ * The `test` function is a wrapper around Mocha's `it` function. It provides a normalized framework for running the
+ * majority of your test assertions inside of a smart-contract, using `DS-Test`.
+ * The API can be used as follows:
+ * ```js
+ * test("<test_name>", "<contract_function_name>")
+ * test("<test_name>", ["<contract_function_name>"])
+ * test("<test_name>", { method: "<contract_function_name>", overrides: {}, shouldFail: false, debug: false })
+ * test("<test_name>", [{ method: "<contract_function_name>", overrides: {}, shouldFail: false, debug: false }])
+ * test("<test_name>", [{ method: "<contract_function_name>", shouldFail: false, debug: false }], {})
+ * ```
+ * All of the above examples are equivalent.
+ * Many contract functions can be called as a part of the same test:
+ * ```js
+ * test("<test_name>", [<test_fn1>, <test_fn2>, <test_fn3>])
+ * ```
+ * Individual test functions can describe their own overrides with the `overrides` property.
+ * If an object is passed in as the third argument to `test`, it will be used as the default overrides for all test
+ * functions.
+ * The following are equivalent:
+ * ```js
+ * test("<test_name>", [{ method: "<contract_function_name>", overrides: { from: "0x123" } }])
+ * test("<test_name>", [{ method: "<contract_function_name>" }], { from: "0x123" })
+ * ```
+ * In the above cases, the `from` override must be a signer.
+ * The `shouldFail` property can be used to indicate that the test function should fail. This should be used sparingly
+ * as it is not possible to match on the failure reason.
+ * Furthermore, the `debug` property can be used to print print any thrown errors when attempting to
```

Comment:
```suggestion
 * Furthermore, the `debug` property can be used to print any thrown errors when attempting to
```

Reply:
We could also refer to README.md instead of repeating this here

Reply:
Yeah, I couldn't decide whether to do a README or inline docs and ended up just not making a decision and doing a little copy-pasta. It kinda looks bad, but in my experience with READMEs like this, they almost always end up outdated and incorrect, whereas inline docs are more likely to get updated (but can still suffer from the same thing without doc-tests). 

Thoughts? I should probably just leave a comment in the README saying the double-check the inline docs and open a PR if the readme is out of date

Reply:
@darioush tagging you just to make sure you see the above

Reply:
I am not too opinionated one way or the other. Your call.

Reply:
Going to leave it for now, just so we can get this in there.